substra/sdk/backends/local/backend.py

```diff
@@ -559,12 +555,14 @@ def _add_traintuple(self, key, spec, spec_options=None):
             key=key,
             creator=_BACKEND_ID,
             algo={
-                "hash": spec.algo_key,
+                "key": spec.algo_key,
+                "hash": algo.content.hash_,
                 "name": algo.name,
                 "storage_address": algo.content.storage_address
             },
             dataset={
-                "opener_hash": spec.data_manager_key,
+                "key": spec.data_manager_key,
+                "opener_hash": spec.data_manager_key,  # TODO: pass actual opener hash
```

Comment:
Above in the function we get `data_manager = self._db.get(schemas.Type.Dataset, spec.data_manager_key)` so you can replace by `data_manager.opener.hash_`